### PR TITLE
Switching underlying PDF viewer

### PR DESCRIPTION
**Pull request by mkoo21**
Switching the pdf renderer from [an old deprecated library](https://github.com/JoanZapata/android-pdfview) to a [newer, better maintained library](https://github.com/barteksc/AndroidPdfViewer) per #34.

Using the newer library will resolve some pixelation issues when zooming in android 6. It also lets you use the Apache License 2.0 rather than GPL.

### CODE REVIEW

**Comment by andrejunges** [comment]
@mkoo21 is it ready to be used/merged?

**Comment by mkoo21** [comment]
I have been using it for a while now without noticing any issues (android only, since react-native-quick-look takes care of document rendering in ios very well)

**Comment by andrejunges** [comment]
I've tested it - and now the app isn't crashing, but nor its rendering the pdf (this behavior only happens for one of the pdf files, but I can open it in other pdf apps or ios).
Im trying now to open it from the URI

**Comment by mkoo21** [comment]
did the old lib work better for you?

**Comment by andrejunges** [comment]
nop, I got a crash in that specific file in the old lib (PDF is corrupted..)

**Comment by andrejunges** [comment]
any idea/suggestion?

**Comment by mkoo21** [comment]
Haven't run into the issue myself so not really sure what to say. Does it look similar to [this issue](https://github.com/barteksc/AndroidPdfViewer/issues/175)? If there really isn't anything special about your file maybe ask there and see if you get a better response

**Comment by andrejunges** [comment]
Indeed its similar.
I noticed now that the problem is related to the pdf's size.
Files with over 400kb are not opening in here

**Comment by andrejunges** [comment]
@mkoo21 I finally found the problem.
Actually it was in my code. I passed the file path to my componente before the file creation completes.
So it was trying to read the file when it was still creating it (before being closed).
classic IO problem
thanks for the help anyway 👍

**Comment by mkoo21** [comment]
Sure, glad you got it resolved and sorry I wasn't able to help more

**Comment by andrejunges** [comment]
image
Any idea why I get this error when I rerender an pdf on Android?

**Comment by gyzerok** [comment]
@mkoo21 @andrejunges any updates about the PR?

**Comment by mkoo21** [comment]
Issue #39 has nothing to do with this PR (you'll get it with the old lib as well), so it's been the same as always

**Comment by gyzerok** [comment]
@mkoo21 can you rebase your fork on top of current master so I can check if it works? And them we can probably ask to merge it.

**Comment by mkoo21** [comment]
done

**Comment by gyzerok** [comment]
@mkoo21 thank you! Will try it next week and report my results.

**Comment by aeirola** [comment]
Nice work, the new PDF rendering library seems to work better. Found a couple of issues though:

1. The page numbering of the new library seems to be zero-indexed, meaning that a PDF always opens on the second page. Passing the prop `pageNumber={0}` doesn't help, since there is a check for non-positive page numbers at https://github.com/cnjon/react-native-pdf-view/blob/master/android/src/main/java/com/keyee/pdfview/PDFViewManager.java#L106

2. The old dependency is still present in the gradle file, causing unneccessary libraries to be loaded in the application. Would be nice if this could be removed.

**Comment by gyzerok** [comment]
@mkoo21 sorry for long response. Here my coworker tried your branch and reported some issues.

**Comment by mkoo21** [comment]
Thanks for the feedback, will take a look when I get home

**Comment by gyzerok** [comment]
Hello @mkoo21! Do you have any updates?

**Comment by mkoo21** [comment]
Yes, sorry, I added some minor changes as requested but haven't really had a chance to test them yet. You are welcome to try these 'unstable' changes if you would like, otherwise I will hopefully get back to you by the end of the week

**Comment by mkoo21** [comment]
Yes, sorry, I added some minor changes as requested but haven't really had a chance to test them yet. You are welcome to try these 'unstable' changes if you would like, otherwise I will hopefully get back to you by the end of the week

**Comment by aeirola** [comment]
Thanks for the changes, seem to work nicely. I think we'll take these changes in use in our application. Hopefully they will soon be merged into the main repository too.

File: android/src/main/java/com/keyee/pdfview/PDFViewManager.java [hunk 1]
```diff
@@ -109,7 +109,7 @@ public void setAsset(PDFView view, String ast) {
     @ReactProp(name = "pageNumber")
     public void setPageNumber(PDFView view, Integer pageNum) {
         //view.setPageNumber(pageNum);
-        if (pageNum > 0){
+        if (pageNum > -1){
```
Comment (gyzerok) [inline android/src/main/java/com/keyee/pdfview/PDFViewManager.java]: Would it be clearer to do `pageNum >= 0`?

Comment (mkoo21) [inline android/src/main/java/com/keyee/pdfview/PDFViewManager.java]: Sure, I can do that if it helps

**Comment by gyzerok** [comment]
@sibelius it seems like we haven't had any issues while using version in this PR. Should we maybe merge it then?

**Comment by sibelius** [comment]
@gyzerok could u please upgrade the example project to the latest version of React Native?

So I can test against your branch and merge it

**Comment by gyzerok** [comment]
@sibelius oh, it's not my PR.

@mkoo21 could you, please, do what @sibelius is asking for?

**Comment by mkoo21** [comment]
updated example

**Comment by sibelius** [comment]
it does look good, thanks for the awesome work

**Pull request merged**